config/hubs/cloudbank.cluster.yaml

```diff
@@ -350,3 +350,38 @@ hubs:
                 - akonrad@mills.edu
                 - wang@mills.edu
               username_pattern: '^(.+@mills\.edu|yuvipanda@gmail\.com|choldgraf@gmail\.com|georgiana\.dolocan@gmail\.com|aculich@berkeley\.edu|jpercy@berkeley\.edu|deployment-service-check)$'
+  - name: palomar
+    domain:
+      - palomar.cloudbank.2i2c.cloud
+    template: basehub
+    auth0:
+      connection: google-oauth2
+    config:
+      jupyterhub:
+        homepage:
+          templateVars:
+            org:
+              name: Palomar Community College
+              logo_url: https://www2.palomar.edu/pages/guidelines/files/2016/03/webGuide_screened_2-color_light.jpg
+              url: https://www2.palomar.edu/
+            designed_by:
+              name: 2i2c
+              url: https://2i2c.org
+            operated_by:
+              name: CloudBank
+              url: http://cloudbank.org/
+            funded_by:
+              name: CloudBank
+              url: http://cloudbank.org/
+        hub:
+          config:
+            Authenticator:
+              admin_users: &palomar_admins
+                - yuvipanda@gmail.com
+                - choldgraf@gmail.com
+                - georgiana.dolocan@gmail.com
+                - aculich@berkeley.edu
+                - sean.smorris@berkeley.edu
+                - tcanon@palomar.edu
+                - PChen@palomar.edu
+              username_pattern: '^(.+@palomar\.edu|yuvipanda@gmail\.com|choldgraf@gmail\.com|georgiana\.dolocan@gmail\.com|aculich@berkeley\.edu|sean.smorris@berkeley\.edu|deployment-service-check)$'
```

Comment:
Looking at it some more, I think it's a regular expression that permits everyone from `@palomar.edu`, but also these explicitly listed addresses otherwise.

Reply:
When an admin should not be allowed? I mean... I was expecting the admins to be excluded from the `username_pattern`...

Reply:
> I was expecting the admins to be excluded from the username_pattern

Me too :eyes: 

This is coming from `jupyterhub`. All users need to comply with `username_pattern`, including admins. Some code references:
- The [`username_pattern` is checked in `validate_user`](https://github.com/jupyterhub/jupyterhub/blob/6be3160d7487bb362df7095f24454ef2814951fc/jupyterhub/auth.py#L240-L241`)
- [`validate_user` is called](https://github.com/jupyterhub/jupyterhub/blob/6be3160d7487bb362df7095f24454ef2814951fc/jupyterhub/auth.py#L485-L487) no matter the user type

Other thing that I noticed and I think it's related, is about `allowed_users`. All the admins, apart from being listed in the `admin_users` list, need to be in this allowed list too [otherwise they will be denied access](https://github.com/jupyterhub/jupyterhub/blob/6be3160d7487bb362df7095f24454ef2814951fc/jupyterhub/auth.py#L492-L511).
In my mind, if the user it's an admin, then they should be allowed by default.

(These over-complicates a bit https://github.com/2i2c-org/pilot-hubs/pull/311 too)

I'm thinking about opening a PR on jupyterhub's side about this, but I would like to know your opinion first, in case my intuition is not right and I'm trying to create an anti-pattern.

Reply:
Jumping in a bit late here. 

I did look around at entries for the other hubs in the cloudbank.cluster.yaml file. The username_pattern element was in the mills definition but not in the others. I probably should have checked out whether or not to use what appeared to be the general pattern:

               allowed_users: &sbcc_users 
                - yuvipanda@gmail.com
                - choldgraf@gmail.com
                - georgiana.dolocan@gmail.com
                - ericvd@gmail.com
                - sean.smorris@berkeley.edu
                - nfguebels@pipeline.sbcc.edu
              admin_users: *sbcc_users

as opposed to the pattern mills is using:

               admin_users: &mills_admins
                - yuvipanda@gmail.com
                - choldgraf@gmail.com
                - georgiana.dolocan@gmail.com
                - aculich@berkeley.edu
                - jpercy@berkeley.edu
                - akonrad@mills.edu
                - wang@mills.edu
              username_pattern:'^(.+@mills\.edu|yuvipanda@gmail\.com|choldgraf@gmail\.com|georgiana\.dolocan@gmail\.com|aculich@berkeley\.edu|jpercy@berkeley\.edu|deployment-service-check)$'

Are we ok here? Or should this change?